Log actions on modules : install, uninstall, upgrade

jf-viguier opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project!

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/#pull-requests

For type and category see:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#type--category
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.0.x
| Description?      | Log actions on modules : install, unistall, upgrade
| Type?             | new feature
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Install/unistall/upgrade a module and check logs
| UI Tests | https://github.com/florine2623/testing_pr/actions/runs/11894798451 ✅ 
| Fixed issue or discussion?     | Fixes https://github.com/PrestaShop/PrestaShop/discussions/37386
| Sponsor company   |Creabilis.com

jf-viguier committed:
Log actions on modules : install, unistall, upgrade

ps-jarvis commented:
<!-- PR_WORDING -->
This pull request seems to contain new translation strings. I have summarized them below to ease up review:

- `Admin.Modules.Notification`
    - [ ] `Starting module install: %s v%s`
    - [ ] `Module installed succesfully: %s v%s`
    - [ ] `Starting module upgrade: %s v%s to v%s`
    - [ ] `Starting module uninstall: %s v%s`
    - [ ] `Module uninstalled succesfully: %s v%s`

(Note: this is an automated message, but answering it will reach a real human)

Hlavtox commented on classes/module/Module.php:
```diff
@@ -473,6 +484,16 @@ public function install()
         Group::addRestrictionsForModule($this->id, Shop::getShops(true, null, true));
+        PrestaShopLogger::addLog(
+            Context::getContext()->getTranslator()->trans(
+                'Module installed succesfully: %s v%s',
```
```suggestion
                'Module installed successfully: %s v%s',
```

jf-viguier replied:
@Hlavtox thanks, it's fixed

Hlavtox commented on classes/module/Module.php:
```diff
@@ -632,6 +663,16 @@ public function runUpgradeModule()
             Module::upgradeModuleVersion($this->name, $upgrade['upgraded_to']);
+            PrestaShopLogger::addLog(
+                Context::getContext()->getTranslator()->trans(
+                    'Module upgraded succesfully: %s to v%s',
```
```suggestion
                    'Module upgraded successfully: %s to v%s',
```

Hlavtox commented on classes/module/Module.php:
```diff
@@ -843,7 +894,16 @@ public function uninstall()
-
+            PrestaShopLogger::addLog(
+                Context::getContext()->getTranslator()->trans(
+                    'Module uninstalled succesfully: %s v%s',
```
```suggestion
                    'Module uninstalled successfully: %s v%s',
```

Progi1984 commented:
@jf-viguier Why not 9.0.x ?

jf-viguier committed:
add another logging on module upgrade.

jf-viguier commented:
Thanks  @ShaiMagal @Hlavtox for your approve

florine2623 reviewed (dismissed):
Hello @jf-viguier ,

I performed install/uninstall/upgrade actions, and here's what I have in the logs :

Upgrade
<img width="1065" alt="Screenshot 2024-11-19 at 14 22 08" src="https://github.com/user-attachments/assets/edf66a14-d6ab-450b-94f5-ae6e490fa34c">

Install 
<img width="1106" alt="Screenshot 2024-11-19 at 14 21 53" src="https://github.com/user-attachments/assets/7111eb41-b37e-4ecd-ad7b-69b067573ec1">

Uninstall
<img width="1097" alt="Screenshot 2024-11-19 at 14 21 42" src="https://github.com/user-attachments/assets/7f790e86-9db5-43bf-84b5-0bc5dced9a98">


Uninstall the folder as well
<img width="1097" alt="Screenshot 2024-11-19 at 14 21 29" src="https://github.com/user-attachments/assets/9d9cd5a3-71fe-4e15-b8d1-8f6192f5d607">

Works as expected. It is QA ✅ 
Thanks!

Hlavtox reviewed (changes requested):
@florine2623 Thank you for QA, but we have some red tests.

So, @jf-viguier can you please resolve the test issues so we can merge it? :-)

jf-viguier committed:
Use legacy PrestaShopLogger adapter: LegacyLogger

jf-viguier committed:
Allow log duplicate: a module can be installed several times, we want all logs

jf-viguier commented:
@Hlavtox I've used the adapter and made a little improvment to allow log duplicate: a module can be installed several times, we want all logs.
@florine2623 thanks to test again, should be ok.

jf-viguier committed:
change LegacyLogger to pass tests

jf-viguier commented:
I'm searching for help on fixing test error

kpodemski commented:
@jf-viguier try changing `ConfigurationCore` and replace `if ($results) {` to `if (is_array($results) && !empty($results)) {`

jf-viguier committed:
rework logging to allow to mock log for unit test

jolelievre commented on classes/module/Module.php:
```diff
@@ -373,6 +373,19 @@ public function __construct($name = null, ?Context $context = null)
+                [$this->name, $this->version],
+                'Admin.Modules.Notification'
+            ),
+            1,
```
```suggestion
            PrestaShopLogger::LOG_SEVERITY_LEVEL_INFORMATIVE
```

jolelievre replied:
Can you please use the const everywhere so we can understand right away the level of severity?

jf-viguier replied:
Done thnks @jolelievre

jolelievre commented on classes/module/Module.php:
```diff
@@ -473,6 +486,18 @@ public function install()
+            1,
+            null,
+            'Module',
+            null,
```
Why not including the module ID here for the actions where it's available?

jf-viguier replied:
Because the id is not a relevant information : it change at every install/unintall. Module name is a better identifier.

jolelievre replied:
Yes you're right, too bad object_id is forced to an int, we could have saved the name if it was a string

jolelievre commented on tests/Unit/Core/Module/ModuleManagerTest.php:
```diff
@@ -77,10 +77,11 @@ public function setUp(): void
             ->getMock()
         ;
         $this->moduleManager->method('upgradeMigration')->willReturn(true);
+        $this->moduleManager->method('log')->willReturn(true);
```
NOt sure the mock is adapted since the method doesn't return anything

jf-viguier replied:
removed, thanks

jolelievre commented on src/Core/Module/ModuleManager.php:
```diff
@@ -229,6 +231,15 @@ public function upgrade(string $name, $source = null): bool
 
         $this->hookManager->exec('actionBeforeUpgradeModule', ['moduleName' => $name, 'source' => $source]);
 
+        $this->log(
```
Why is this the only log you added in the module manager while the others are all in the legacy class? Is this because you need to log the beginning and the end of the upgrade and it was the only way to do this? 🤔

jf-viguier replied:
@jolelievre because the module upgrade process is there and not in the legacy class. Install/uninstall are in the legacy class.

jolelievre replied:
Hmm it's not exactly true, you still have those methods in the legacy class:
- initUpgradeModule
- runUpgradeModule
- upgradeModuleVersion
- onUpgrade

You even added logs in the `runUpgradeModule`, that's why I wondered if it could all be in the legacy class only?

jf-viguier replied:
runUpgradeModule in legacy module is not called by the controller, that' why I've added it in ModuleManager

jf-viguier committed:
log severity and mock impr

jolelievre reviewed (dismissed):
Thanks @jf-viguier

jf-viguier committed:
move module upgrade logging to legacy class as suggested by Jo. Support module install logging not on disk

jf-viguier commented:
I've moved module upgrade logging to legacy class as suggested by @jolelievre.
I've added support to module install logging not on disk via distribution API (initUpgradeModule launched in this case)

jolelievre reviewed (dismissed):
Thanks @jf-viguier

florine2623 reviewed (commented):
Hello @jf-viguier ,

Relaunched the auto tests : https://github.com/florine2623/testing_pr/actions/runs/11969128394

Retested your PR
Works as expected for actions install and uninstall, but not for upgrade.

I can't upgrade my module, I have an exception : 

<img width="1053" alt="Screenshot 2024-11-22 at 10 04 04" src="https://github.com/user-attachments/assets/005c2a7b-2eaa-4b35-bd6b-3b86bcca6d40">

Could you check ?
Thanks!

kpodemski commented:
@florine2623 I don't think it's related to this PR, probably to recent changes for the API, @jolelievre is that possible? 🤔

jf-viguier commented:
@florine2623 I don't have this exception on upgrade. It seems not related to my pr

jolelievre commented:
Since the new logs are related to module actions it MAY be related I'll look into it on monday, but in the meantime it May be worth trying to relaunch the tests as we regularly have some random failures

jf-viguier commented:
@jolelievre @florine2623 have you fixed your route issue ?

jolelievre commented:
@kpodemski we didi implement new endpoints for modules in the new API based on CQRS but they are not integrated in the BO yet (it would be nice to do this in v9.1 or at least to refacto the controller so they are based on CQRS implementation). So I don't think the recent changes impacted this, and neither should this PR

jf-viguier commented:
@jolelievre conflict fixed

jf-viguier commented:
@florine2623 can you please test again ?

jf-viguier commented:
thanks @Hlavtox

jf-viguier commented:
Thanks @kpodemski

florine2623 reviewed (approved):
Hello @jf-viguier ,

I relaunched the automated tests as some edits were made : https://github.com/florine2623/testing_pr/actions/runs/12274625341 
Tests red not due to this PR, it will be fixed by automation team ^^

Retested the PR.

Log actions are well displayed.
No exception displayed when trying to upgrade a module.

<img width="649" alt="Screenshot 2024-12-11 at 14 00 43" src="https://github.com/user-attachments/assets/fbf2ec60-b08d-41a6-b742-b6b12769a8be">
<img width="618" alt="Screenshot 2024-12-11 at 14 02 14" src="https://github.com/user-attachments/assets/d7804a4f-ab7d-4bb4-a581-b9c4a2619449">
<img width="629" alt="Screenshot 2024-12-11 at 14 02 52" src="https://github.com/user-attachments/assets/4de6509a-b951-4fea-a4fe-7ae8b144b08e">


LGTM.
It is QA ✅ 
Thanks!

matthieu-rolland commented:
thank you @jf-viguier & @florine2623
